python/GafferArnold/ArnoldTextureBake.py

```diff
@@ -84,13 +85,27 @@ def __init__( self, name = "__CameraSetup" ) :
 			self["__chunkExpression"].setExpression( inspect.cleandoc(
 				"""
 				import collections
+				import re
+
 				rawInfo = parent["__udimQuery"]["out"]
 
 				defaultFileName = parent["defaultFileName"]
 				defaultResolution = parent["defaultResolution"]
 
+				selectUdimsStr = parent["selectUdims"]
+
+				# FrameList really ought to take care of this check, instead of just doing
+				# something obviously wrong
+				if re.match( ".*[0-9] +[0-9].*", selectUdimsStr ):
+					raise RuntimeError( "ArnoldTextureBake : Udim list must be comma separated." )
+
+				selectUdims = set( [str(i) for i in IECore.FrameList.parse( selectUdimsStr ).asList()] )
```

Comment:
Sanity check (my sanity, not yours) : `str( i )` is OK and we don't need to worry about padding, because all UDIMs start at 1000?

Reply:
It's true that starting at 1001 is certainly the convention, but it doesn't hurt to not fail on lower ones.  I had originally gone with converting int to string because it obviously can't fail, but on reconsideration, that UDIM strings we're working with came from UdimQuery converting int to string, so converting everything to int should be safe, and little bit more succinct, and handles this case OK, so I've gone with that.